lib/features/auth/data/repositories/auth_repo_impl.dart

```diff
@@ -0,0 +1,23 @@
+import 'package:flowery_e_commerce/core/networking/common/api_result.dart';
+import 'package:flowery_e_commerce/features/auth/data/data_sources/contracts/offline_data_source.dart';
+import 'package:flowery_e_commerce/features/auth/domain/contracts/auth_repo.dart';
+import 'package:flowery_e_commerce/features/auth/domain/entities/request/login_request_entity.dart';
+import 'package:flowery_e_commerce/features/auth/domain/entities/response/login_response_entity.dart';
+import 'package:injectable/injectable.dart';
+
+import '../data_sources/contracts/auth_online_data_source.dart';
+
+@Injectable(as: AuthRepository)
+class AuthRepositoryImpl implements AuthRepository {
+  final AuthOnlineDataSource _onlineDataSource;
+
+  @factoryMethod
+  AuthRepositoryImpl(this._onlineDataSource);
+
+  @override
+  Future<ApiResult<LoginResponseEntity>> login(
```

Comment:
Refactor ApiResult class to DataResult to can use it at online data source and offline data source